Yosemite/Yosemite/Stores/ProductReviewStore.swift

```diff
@@ -158,14 +157,13 @@ private extension ProductReviewStore {
 
     /// Deletes any Storage.ProductReview with the specified `siteID` and `reviewID`
     ///
-    func deleteStoredProductReview(siteID: Int64, reviewID: Int64) {
-        let storage = storageManager.viewStorage
-        guard let productReview = storage.loadProductReview(siteID: siteID, reviewID: reviewID) else {
-            return
-        }
-
-        storage.deleteObject(productReview)
-        storage.saveIfNeeded()
+    func deleteStoredProductReview(siteID: Int64, reviewID: Int64, onCompletion: @escaping () -> Void) {
+        storageManager.performAndSave({ storage in
+            guard let productReview = storage.loadProductReview(siteID: siteID, reviewID: reviewID) else {
+                return
```

Comment:
unrelated to this PR but while reading the code, I wonder if it's worth adding a log here?

Reply:
Good point! Adding a log would help us know when there's a discrepancy between the view context and the background context. It would be nicer if we can add tracking here, but Tracks is not accessible from Yosemite. I added the log in 73ef9737350c0fa9f7f8dfc7c9c55c864e830bf5.